src/librustdoc/html/format.rs

```diff
@@ -456,15 +456,26 @@ impl clean::GenericArgs {
     }
 }
 
-crate fn href(did: DefId, cx: &Context<'_>) -> Option<(String, ItemType, Vec<String>)> {
+// Possible errors when computing href link source for a `DefId`
+crate enum HrefError {
+    // `DefId` is in an unknown location. This seems to happen when building without dependencies
+    // but a trait from a dependency is still visible
+    UnknownLocation,
+    // Unavailable because private
+    Unavailable,
```

Comment:
That implies this should never be hit with --document-private-items, right?
```suggestion
    /// This can only happen for non-local items when `--document-private-items` is not passed.
    Private,
```